hardinfo/pci_util.c

```diff
@@ -20,6 +20,80 @@
 
 #include "hardinfo.h"
 #include "pci_util.h"
+#include "util_ids.h"
+
+gchar *pci_ids_file = NULL;
+
+void find_pci_ids_file() {
+    if (pci_ids_file) return;
+    char *file_search_order[] = {
+        g_strdup("/usr/share/hwdata/pci.ids"),
+        g_build_filename(g_get_user_config_dir(), "hardinfo", "pci.ids", NULL),
+        g_build_filename(params.path_data, "pci.ids", NULL),
+        NULL
+    };
+    int n;
+    for(n = 0; file_search_order[n]; n++) {
+        if (!pci_ids_file && !access(file_search_order[n], R_OK))
+            pci_ids_file = file_search_order[n];
+        else
+            g_free(file_search_order[n]);
+    }
+}
+
+char *pci_lookup_ids_vendor_str(uint32_t id) {
+    gchar *ret = NULL;
+
+    ids_query_result result = {};
+    gchar *qpath;
+
+    if (!pci_ids_file)
+        find_pci_ids_file();
+
+    qpath = g_strdup_printf("%04x", id);
+    scan_ids_file(pci_ids_file, qpath, &result, -1);
```

Comment:
`pci_ids_file` can be NULL here if `find_pci_ids_file()` fails to find in all of the 3 locations. Might be a good idea to `if (!pci_ids_file) return NULL;` after calling `find_pci_ids_file()`.

Reply:
Yeah, but scan_ids_file() will handle it alright. It would just do some useless work making the request.